packages/compose/README.md

```diff
@@ -165,6 +175,22 @@ _Parameters_
 -   _callback_ `Function`: Shortcut callback.
 -   _options_ `WPKeyboardShortcutConfig`: Shortcut options.
 
+<a name="useLazyRef" href="#useLazyRef">#</a> **useLazyRef**
```

Comment:
That one is a bit confusing, it's very unclear what it does and when to use it. Makes me wonder if it deserves to be a public API on the compose package.

Reply:
In my own experience, I find `useLazyRef` very useful in many places though. Whenever we want to create a ref, but only want to create it lazily once, this hook is the way to go.

This hook is basically giving `useRef` the same super power of `useState` function initialization.

```js
useState(() => expensiveCalculation())[0]; // returns a value

useLazyRef(() => expensiveCalculation()); // returns a ref
```

Reply:
"Lazy" sounds a bit weird. Could we call it `useMemoizedRef` or something like that?  I don't see much benefit though. `useRef( useMemo() )` is the same thing?

Reply:
Sounds good. How can we make it clearer that the laziness only applies to initial value?

Reply:
Makes me wonder if there's ever a use case for the initial ref to be set on every render...

Reply:
Nope, not that I can think of 😅. Ref is always only be set on the initial render, it's like the instance variable of class components.

Reply:
Right, so I wonder why we don't do this by default (or why React doesn't).

Reply:
I think it's because that function initializer still adds some overheads to the component, even though it's really small, it will still be used by a large number of projects. `useRef(value)` is also way simpler as a public API too.